sympy/crypto/crypto.py

```diff
@@ -3324,3 +3327,123 @@ def decipher_bg(message, key):
         orig_msg += (m ^ b)
 
     return orig_msg
+
+################# Columnar Transposition Cipher #################
+
+def encipher_columnar(message, key, padding=True):
+    """
+    Performs Regular Columnar Encryption on plaintext and returns ciphertext
+
+    Examples
+    ========
+
+    >> from sympy.crypto.crypto import encipher_columnar
+    >> message = "HELLO WORLD"
+    >> encipher_columnar(message,"SYM")
+    "LWLEHLODEORY"
+
+    Parameters
+    ==========
+
+    message : string, the message to be encrypted.
+    key : string, the key to be used for encryption.
+    padding: bool, decides if random letters are to be padded
+            in the end of message. By default, its value is true.
+
+    Returns
+    =======
+
+    The Encrypted message.
+
+    References
+    ==========
+    https://en.wikipedia.org/wiki/Transposition_cipher#Columnar_transposition
+
+    """
+    message =  ''.join(alphabet for alphabet in message if alphabet.isalpha())
+    key =  ''.join(alphabet for alphabet in key if alphabet.isalpha())
+    n = len(key)
+    words = len(message)
+    remainder = words%n
+
+    if len(message) < 2:
+        return message
+
+    if remainder != 0 and padding == True:
+        message += ''.join(choice(string.ascii_lowercase) for i in range(n - remainder))
+
+    d = defaultdict(list)
+    low = key.lower()
+    for i,c in enumerate(low):
+        d[c].append(i)
+    index_list = [j for i,j in sorted(zip([d[c].pop(0) for c in sorted(low)], range(n)))]
+
+    cipher = ''.join([message[j].upper() for i in range(n) for j in range(index_list.index(i), len(message), n)])
+    return cipher
+
+def decipher_columnar(cipher, key):
+    """
+    Performs Regular Columnar Decryption on ciphertext and returns plaintext
+
+    Examples
+    ========
+
+    >>> from sympy.crypto.crypto import decipher_columnar
+    >>> cipher = "LWLEHLODEORY"
+    >>> decipher_columnar(cipher,"SYM")
+    'HELLOWORLDYE'
+
+    In HELLOWORLDYE, Y and E are the random letters added to the original message (as per the algorithm)
+    and are to be ignored
+
+    Parameters
+    ==========
+
+    cipher : string, the cipher text to be decrypted.
+    key : string, the key to be used for decryption.
+
+    Returns
+    =======
+
+    The Decrypted message.
+
+    References
+    ==========
+    https://en.wikipedia.org/wiki/Transposition_cipher#Columnar_transposition
+
+    """
+
+    cipher =  ''.join(alphabet for alphabet in cipher if alphabet.isalpha())
```

Comment:
I would recommend some variable name changes

alphabet -> c (for character)
words -> N
col_length -> rows

Also, I would recommend looking at the other crypto routines. I think it would be better to have a no-frills column transposition which does nothing but fill and unfill using an unchanged key (e.g. not changing to lower case to get the index and perhaps requiring a permutation list of numbers instead of a string for a key) and then have driver routines that perform the "textbook" encipher/decipher routines (converting to upper case, converting a key to a permutation number, etc...).

It might be good to extract the `index_list` finding routine into a helper `permutation_from_key`. 

If this is supposed to be for learning purposes then we should provide a good building block, not one of many possible final solutions.

Reply:
Ok. I'll get on it.